ecs-logs-collector.sh

```diff
@@ -584,6 +585,21 @@ enable_ecs_agent_debug() {
   esac
 }
 
+get_docker_goroutines() {
+  try "dump docker goroutine stacks"
+  dstdir="${info_system}/docker_stacks"
+  mkdir -p "${dstdir}"
+  # send SIGUSR1 to dump goroutines to
+  # /var/run/docker/goroutine-stacks-<date>.log
+  if kill -SIGUSR1 "$(pidof dockerd)" ; then
+    # find file and copy to our output
+    until find /var/run/docker/ -cmin -1 -type f -name "goroutine-stacks*.log" -exec cp -p '{}' "$dstdir" \; ; do
```

Comment:
I wasn't able to get this to work on every run, `find` exits with a `0` status even when there's no matches. With a `grep` added, I could reliably get this to capture the "status" of the goal we have in mind:

```suggestion
    until find /var/run/docker/ -cmin -1 -type f -name "goroutine-stacks*.log" -exec cp -p '{}' "$dstdir" \; -print | grep -q '.'; do
```

This checks that `find` found (and printed) something when it ran. This should work with and without `set -o pipefail` - can you verify the same?

I used a modified script to check this: https://gist.github.com/jahkeup/010e62840c0d6bf6cb6f191909cc46ae

Reply:
Good call. If `set -o pipefail` is set it will fail if there are multiple matching files because find will keep writing to the pipe after grep exits.